exporter/otlphttpexporter/otlp.go

```diff
@@ -75,14 +76,16 @@ func newExporter(cfg config.Exporter, logger *zap.Logger) (*exporter, error) {
 
 	// client construction is deferred to start
 	return &exporter{
-		config: oCfg,
-		logger: logger,
+		config:    oCfg,
+		logger:    logger,
+		buildInfo: buildInfo,
 	}, nil
 }
 
 // start actually creates the HTTP client. The client construction is deferred till this point as this
 // is the only place we get hold of Extensions which are required to construct auth round tripper.
 func (e *exporter) start(_ context.Context, host component.Host) error {
+	e.config.HTTPClientSettings.WithDefaultUserAgent(exporterhelper.DefaultUserAgent(e.buildInfo))
```

Comment:
Should we make this user configuration? Not sure if this should be enforced.

Reply:
Currently, the user-agent header for HTTP and gRPC exporters falls back to their respective defaults (e.g. `Go-http-client/1.1` `grpc-go/1.40.0`). Users can't opt out of that. (Though they can override it for HTTP requests - https://github.com/open-telemetry/opentelemetry-collector/pull/3970#issuecomment-916372313)

To me, this change updates that default fallback header to be more specific, which doesn't seem like it ought to be configurable, beyond the existing behavior of users providing a custom user-agent header.

WDYT?

Reply:
Indeed, I'd go further and say that this default user agent should always be used by the HTTP and gRPC clients provided by the collector unless an alternative is explicitly set by the user, not just in the exporters that opt into it.  I think that might be complicated by the lack of some of the required information in the clients, `BuildInfo` in particular.